File: Assets/scenes/cometScene/scripts/CometBehaviour.js
#pragma strict




public var velocity : Vector3;
public var rotSpeed : Vector3;

private var uniObj : UniverseObject;


function Start () {
	uniObj = GetComponent.<UniverseObject>();
}

function FixedUpdate(){
	var mod : float = 1f;
	if(uniObj.inDetailSpace){
		mod = 1f;
	} else {
		mod = 0.02f;
	}
	transform.position += transform.TransformDirection(velocity * mod); 
	transform.rotation = transform.rotation * Quaternion.Euler(rotSpeed);

}


function OnCollisionEnter(c : Collision){
	if(c.gameObject.name == "TheShip"){
		//we whomped the player, blow them up
		var theShip = GameObject.Find("TheShip");
		theShip.GetComponent.<ShipCore>().damageShip(1000, "Unplanned collision with Comet Surface");
	} 
		
}

File: Assets/scenes/cometScene/scripts/CometBehaviour.cs
using UnityEngine;
using System;



public class CometBehaviour:MonoBehaviour{
	
	
	public Vector3 velocity;
	public Vector3 rotSpeed;
	
	UniverseObject uniObj;
	
	
	public void Start() {
		uniObj = GetComponent<UniverseObject>();
	}
	
	public void FixedUpdate(){
		float mod = 1f;
		if(uniObj.inDetailSpace){
			mod = 1f;
		} else {
			mod = 0.02f;
		}
		transform.position += transform.TransformDirection(velocity * mod); 
		transform.rotation = transform.rotation * Quaternion.Euler(rotSpeed);
	
	}
	
	
	public void OnCollisionEnter(Collision c){
		if(c.gameObject.name == "TheShip"){
			//we whomped the player, blow them up
			GameObject theShip = GameObject.Find("TheShip");
			StartCoroutine(theShip.GetComponent<ShipCore>().damageShip(1000.0f, "Unplanned collision with Comet Surface"));
		} 
			
	}




}
